roles/docker/tasks/systemcontainer_docker.yml

```diff
@@ -114,6 +114,14 @@
     dest: "{{ docker_systemd_dir }}/custom.conf"
     src: systemcontainercustom.conf.j2
 
+# Set local versions of facts that must be in json format for daemon.json
+# NOTE: When jinja2.9+ is used the daemon.json file can move to using tojson
+- set_fact:
+    l_docker_insecure_registries: "{{ docker_insecure_registries | to_json }}"
+    l_docker_log_options: "{{ docker_log_options | to_json }}"
+    l_docker_additional_registries: "{{ docker_additional_registries | to_json }}"
+    l_docker_blocked_registries: "{{ docker_blocked_registries | to_json }}"
```

Comment:
We probably need default values for the variables in case they're not defined.
```
- set_fact:
    l_docker_insecure_registries: "{{ docker_insecure_registries | default(None, true) | to_json }}"
    l_docker_log_options: "{{ docker_log_options | default(None, true) | to_json }}"
    l_docker_additional_registries: "{{ docker_additional_registries | default(None, true) | to_json }}"
    l_docker_blocked_registries: "{{ docker_blocked_registries | default(None, true) | to_json }}"
```